src/scanners/base.js

```diff
@@ -18,11 +18,16 @@ export default class BaseScanner {
     return 'string';
   }
 
+  static get scannerName() {
+    throw new Error('scannerName is not implemented');
```

Comment:
This seems like something that should be checked during class initialisation, not during access.